plugins/deployment_manager/DeploymentManager.ts

```diff
@@ -132,6 +132,28 @@ export class DeploymentManager {
     );
   }
 
+  /* Deploys missing contracts from the deployment, using the user-space deploy.ts script */
+  async deployMissing(force: boolean = true) {
+    // XXX if this is idempotent we can just always deploy, and do the same for dev
+    //  as is, won't handle cases where the deploy script adds roots or partial redeploys
+    // if force or there are no roots, deploy
+    //  force will also have to change with idempotent deploy changes
+    //   its here for deploy task, which doesn't really care if roots exists or not
+    //    but we'll want another way to specify how idempotent should work
+    const roots = await this.getRoots();
+    if (force || roots.size == 0) {
+      // XXX noted above but cache is a misnomer since we have non-cache files its kind of managing
```

Comment:
Cache here has always been "in-memory or disk-synced filesystem-like thing." The name came because it was a write-through cache (sometimes), or just an in-memory KV store. The key idea is the calling code never needed to worry if it was a simulation (in-memory only) or disk-synced (e.g. for deployments, etc)